packages/web3-providers/src/Store/index.ts

```diff
@@ -0,0 +1,21 @@
+import { type StoreAPI } from '../types/Store.js'
+
+export abstract class StoreProvider<T> implements StoreAPI.Provider<T> {
```

Comment:
in case you're referring to React's API (which also has a getSnapshot), you cannot store mutable objects in it, or you have to do a full-clone in the getSnapshot

Reply:
It's consumed by `useExternalStore`. I stored an instance of this class, which is created once and doesn't change. The one thing that might break the rule is that the instance contains a mutable map. However, since this is used internally, I think it's fine.

Reply:
> So each time the getSnapshot() is invoked, a new instance of Store class should be returned?

No, it should return what it stored. The store itself is untouched.